File: 2537-count-the-number-of-good-subarrays/2537-count-the-number-of-good-subarrays.py
class Solution:
    def countGood(self, arr: List[int], k: int) -> int:
        res = cur = i = 0
        freq = Counter()
        for j in range(len(arr)):
            k -= freq[arr[j]]
            freq[arr[j]] += 1
            while k <= 0:
                freq[arr[i]] -= 1
                k += freq[arr[i]]
                i += 1
            res += i
        return res
        

File: 2537-count-the-number-of-good-subarrays/2537-count-the-number-of-good-subarrays.cpp
class Solution { 
public: 
    long long countGood(vector<int>& arr, int k) {
        long long res = 0;
        unordered_map<int, int> freq;
        for (int i = 0, j = 0; j < arr.size(); ++j) {
            k -= freq[arr[j]]++;
            while (k <= 0)
                k += --freq[arr[i++]];
            res += i;
        }
        return res;
    }
};